### PR TITLE
Various fixes

### PR DESCRIPTION
**Pull request by hypeJunction**

### CODE REVIEW

**Review by jeabakker — APPROVED** [review]

File: mod/pages/start.php [hunk 1]
```diff
@@ -419,6 +421,39 @@ function pages_register_db_seeds(\Elgg\Hook $hook) {
...
 
+/**
+ * Move child pages when parent page is deleted
+ *
+ * @elgg_event delete object
+ *
+ * @param \Elgg\Event $event Event
+ * @return void
+ */
+function pages_delete_handler(\Elgg\Event $event) {
```
Comment (jdalsem) [inline mod/pages/start.php]: why not in the class `->delete()`??

Comment (hypeJunction) [reply mod/pages/start.php]: This way someone can stop the delete

Comment (jdalsem) [reply mod/pages/start.php]: the action in this handler should happen on a succesful delete... having it in an event handler could cause issues if someone stops delete after this event...

Comment (hypeJunction) [reply mod/pages/start.php]: Delete event is a mess, and there is no way to deal with entity-dependent clean up. I am leaving it in a handler until there is unified way of doing this. I don't think there is a delete:after event

Comment (jeabakker) [reply mod/pages/start.php]: > I don't think there is a delete:after event

yes there is https://github.com/Elgg/Elgg/blob/master/engine/classes/Elgg/Database/EntityTable.php#L683

Comment (jdalsem) [reply mod/pages/start.php]: if you move it to delete:after it makes more sense.. can merge that

Comment (hypeJunction) [reply mod/pages/start.php]: It wouldn't work, because metadata would have been deleted by then. Moved to the delete method

Comment (jdalsem) [reply mod/pages/start.php]: perfect

**Comment by jdalsem** [comment]
test failure

**Comment by jdalsem** [comment]
tried a restart

**Comment by hypeJunction** [comment]
It keeps failing intermittently with memcache, have no clue why

**Pull request merged**